src/inmanta/server/services/compilerservice.py

```diff
@@ -369,6 +369,32 @@ async def request_recompile(
             LOGGER.info("Skipping compile because server compile not enabled for this environment.")
             return None, ["Skipping compile because server compile not enabled for this environment."]
 
+        async def merge() -> Optional[uuid.UUID]:
+            """
+                Tries merging this compile request with an existing one.
```

Comment:
What is the heuristic used to merge this? I need some more documentation because I cannot derive it based on the code alone.

Reply:
The oldest not-in-progress compile request `c` for which `len(c.metadata) == 0 and c.do_export == do_export and c.environment_variables == env_vars` is selected for merging. `c.force_update` is updated to `c.force_update or force_update`.
I'll update the documentation of this method to reflect this.

Reply:
I realize that the code changed a lot since this question. However, I am not able to find exactly what the rules are to merge two requests.

Reply:
The rules remain largely the same, only `c.metadata` is ignored. See [`_compile_merge_key()`](https://github.com/inmanta/inmanta/pull/2180/files#diff-e870351649d28d5d3113d16d0aa3b0c8R389-R391) and [its caller](https://github.com/inmanta/inmanta/pull/2180/files#diff-e870351649d28d5d3113d16d0aa3b0c8R454-R458).
So when a compile starts, all merge candidates are collected (based on `_compile_merge_key()`). `force_update` is set to `any(c.force_update for c in chain([compile], merge_candidates))` and the compile starts. When it finishes, all merge candidates are marked as completed and the compile report is attached.

Reply:
Can you add this as docs somewhere in the code?